mlflow/utils/_capture_transformers_modules.py

```diff
@@ -0,0 +1,81 @@
+"""
+This script should be executed in a fresh python interpreter process using `subprocess`.
+"""
+import json
+import os
+import sys
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.utils._capture_modules import _CaptureImportedModules, parse_args, store_imported_module
+
+
+class _CaptureImportedModulesForHF(_CaptureImportedModules):
+    """
+    A context manager to capture imported modules by temporarily applying a patch to
+    `builtins.__import__` and `importlib.import_module`.
+    Used for 'transformers' flavor only.
+    """
+
+    def __init__(self, module_to_throw):
+        super().__init__()
+        self.module_to_throw = module_to_throw
+
+    def _wrap_package(self, name):
+        if name == self.module_to_throw or name.startswith(f"{self.module_to_throw}."):
+            raise ImportError(f"Disabled package {name}")
+
+    def _record_imported_module(self, full_module_name):
+        self._wrap_package(full_module_name)
+        return super()._record_imported_module(full_module_name)
+
+    def __enter__(self):
+        self.use_tf = os.environ.get("USE_TF") if "USE_TF" in os.environ else None
+        self.use_torch = os.environ.get("USE_TORCH") if "USE_TORCH" in os.environ else None
+        if self.module_to_throw == "tensorflow":
+            os.environ["USE_TORCH"] = "TRUE"
+        elif self.module_to_throw == "torch":
+            os.environ["USE_TF"] = "TRUE"
+
+        return super().__enter__()
+
+    def __exit__(self, *_, **__):
+        # Revert the patches
+
+        if "USE_TF" in os.environ:
+            if self.use_tf is not None:
+                os.environ["USE_TF"] = self.use_tf
+            else:
+                del os.environ["USE_TF"]
```

Comment:
I'd define a helper function to avoid repetition:

```python
def restore_env_var(key, value):
    if value is None and key in os.environ:
        del os.environ[key]
    else:
        os.environ = value
```